core/src/main/java/org/testcontainers/dockerclient/DockerClientProviderStrategy.java

```diff
@@ -137,10 +140,18 @@ public static DockerClientProviderStrategy getFirstValidStrategy(List<DockerClie
                                 .peek(strategy -> log.info("Loaded {} from ~/.testcontainers.properties, will try it first", strategy.getClass().getName())),
                         strategies
                                 .stream()
-                                .filter(it -> !it.getClass().getCanonicalName().equals(dockerClientStrategyClassName))
-                                .filter(DockerClientProviderStrategy::isApplicable)
                                 .sorted(Comparator.comparing(DockerClientProviderStrategy::getPriority).reversed())
                 )
+                .filter(new Predicate<DockerClientProviderStrategy>() {
```

Comment:
Why can't we use `distinct()` here?

Reply:
It compares by value, while we need to compare by type

Reply:
D'oh, yes!

Reply:
Streams API would benefit from a `distinct(Function<T,R>)` method, it seems 😂